### DISCUSSION TITLE
Getting started with Pandora faster

### DISCUSSION THREAD

**Discussion by bitbrain** [bitbrain/pandora#84] [Proposals]
Pandora by design is generic and does not provide any specific systems like quests or items out of the box. Instead, users can define their own structure how they please, especially since every game is different and may have different requirements. However, starting from scratch over and over is not a good experience, especially when it comes to things like game jams or smaller games.

For example, #83 suggested to add a quest system into this addon. There might be many other examples of data structures (inventory system, material system, combat stats, world economy) that someone would always need to design from scratch. Making that initial experience nicer of getting started could help people to have a "starting point" to go from. This proposal suggests some ideas around speeding up the process to get started with this addon.

# Idea: import data.pandora files

Allow users to define their own structure in Pandora (including custom scripts+icons) and let them export it. Other users can just import it into their project then. #17 solves this partially and it would be a good start.

## Quest example

A user can create their own repo where they have a 'pandora quest model' including additional methods/scripts. Perhaps it could be even its own addon but for now let's keep it simple. Someone can create their quest system and then save the data.pandora file.

Other users then can import that file into their setup (somehow) to have a quest system with data out of the box.

# Idea: Resource migration Wizard

Another idea would be to have a "Migration Wizard" that allows for scanning a folder in `res://` for existing resource files. For each resource file, it could generate categories with fields, as well as scanning `.tres` files for then populating entities. This migration would be a "best effort" approach, so not all fields may be able to be migrated. There needs to be some form message/log informing the user about what fields we could not migrate.

## Item system example

Given I have a bunch of resources such as:

- `class_name Item extends Resource`
- `class_name Ore extends Item`
- `class_name Weapon extends Item`

as well as some saved resources:

- `IronOre.tres`
- `CopperSword.tres`

Running the wizard, Pandora generates the following categories/entity structure:

- Items
   - Ores
      - Iron Ore
   - Weapons
      - Copper Sword

As an optional step, the user could tick a checkbox to also "use resources as scripts" which then modifies the original scripts in place, e.g. in this case, Pandora would go and change `Item` to:

```gdscript
class_name Item extends PandoraEntity
```
and then automatically set the entity scripts on imported entities accordingly.

**Comment by rustednail1991-cyber** [comment]
hai how are u im hoping ur able to help me idk if ur one of pandoras creators but i hope u are i have pandora i want to run ostim.....with sexlab...alsp fnis creature pack just the pack tho not the engine also ostim scenes freeze at start of an animation scene right now i am using both fnis engine and pandora is it possible to just use pandora please say ur able to help
